### ISSUE TITLE
Add ability to copy a district and it Pieces and Civil Parishes to another County 

### ISSUE THREAD

**Issue by PatReynolds** [FreeUKGen/FreeCENMigration#879]
Geoff: This has also proved that we cannot have Civil Parishes like Bitton (GLS) in the PARMS because it will create an error in CSVProc.
Pat: is this because there is no way of knowing which county a Civil Parish is in from the ED number? If so, and I have your solution tied to the correct problem, I agree that an extra column showing the County that a Civil Parish is in would solve this issue. This would also permit searching by county picking up Civil Parishes that were, for census purposes, considered in part of another county. I think the county search should show all the Civil Parishes hierarchically contained in that county and ALSO the Civil Parishes identified as being in the county by the transcriber, and noted as such in the additional column. Rather than having variant spellings of the county (e.g. both Newark and Newark-on-Trent) presented as choices on the search page, the users should be presented with a single choice, that would search both variants.

**Comment by PatReynolds** [comment]
@PatReynolds  to review after fulrough

**Comment by PatReynolds** [comment]
Aplogies, all - overlooked this one. 
It seems to me that a district can have more than one county in it (as with the Yorkshire ones I'm working on now, in districts that straddle Durham and Yorkshire, etc.  So why cann Bitton not have the Chapman code GLS?  

A search for all GLS should return Bitton.  A search for the district or subdistrict Bittom is in should find Bittom.

**Comment by Captainkirkdawson** [comment]
@PatReynolds a district, its pieces and civil parish belong to a single county. A single district cannot be connected to more than one county. However there is nothing to stop having districts of the same name in different counties. The pieces and civil parishes that belong to each district will be the appropriate county. Logically they are separate districts. 
But the extrication done from TNA only creates a single district in one county. Although sometimes the notes do say some civil places belong to a different county. 
It would require a manual restructure to set it up

**Comment by PatReynolds** [comment]
To be done by a Data Manager or System Manager - to be raised case by case, as identified.

Not needed before we implement the new PARMS.

**Comment by Captainkirkdawson** [comment]
The capability to copy a District, it Pieces and Civil Parishes to another County has been implemented on test3.
Together with the ability to delete a District, a Piece and a Civil Parish. This allows the contents of the district to be tailored according to the specific requirements. The functionality is restricted to data managers and system administrators due to it ability to damage the collections.
It need to be tested by specifying a district to be copied to another county and how its pieces and their civil parishes are to be distributed.

**Comment by richpomfret** [comment]
@PatReynolds to identify district and provide @Captainkirkdawson with the bits he requires.

**Comment by geoffj-FUG** [comment]
Kirk
The copy will not fix them all. Take this 1851 piece for instabce (yes, 1851 again!)

1944B | Keynsham | Bitton Holy   Trinity, Brislington, Burnet, Compton Dando, Corston, Keynsham, Newton,   Newton St Loe, Oldland, Priston, Queen Charlton, Saltford, Stanton Prior,   Whitchurch
-- | -- | --

Some of the Civil Parishes are in GLS and some are in SOM.

Some Districts can be copied lock stock and barrel such as 1901. 

2272 | Culmstock | Appledore (Devon), Ashculm   (Devon), Ayshford (Devon), Burlescombe (Devon), Clayhidon (Devon), Culm Davy   (Devon), Culmstock (Devon), Ford (Devon), Garlandhayes (Devon), Hemyock   (Devon), Hillmoor (Devon), Holcombe Rogus(Devon), Millmoor (Devon),   Nicholashayne (Devon), Prescott (Devon), Rosemary Lane (Devon), Upcott   (Devon), Westleigh (Devon)
-- | -- | --

Why Culmstock was ever included in SOM I have no idea. It seems to be a numbering error.

I am sure that if I looked hard enough I would find examples at Sub District (Piece) level as well. 

Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG Sorry I do not follow. 
One copies Keynsham to GLS; One deletes from SOM the Pieces and Civil Parishes that are not in SOM. One goes to GLS and deletes the Pieces and Civil Parishes that are not in GLS.
It is a one time set up.
The GLS coordinator loads the piece/civil parishes for GLS and you do the same for SOM.  

Or is my ignorance of Freecen missing something?

**Comment by geoffj-FUG** [comment]
Kirk

 

OK. Got it!

 

That can creates 2 part of the same Piece. In that case they will need to be numbered HO107_1234A and B or HO107_1234_a and b. That will show that the piece has been split.

 

Geoff

**Comment by geoffj-FUG** [comment]
Just logging another point so that it doesn’t get lost.

 

The procedure will require a transfer of a copy of the relevant image(s) to the other County Coordinator.

 

Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG The key is we must try this on test3 to make sure it functions as expected. 
It might be that the copy needs augment the piece number so that they are different

**Comment by geoffj-FUG** [comment]
Kirk

 

Can I upgrade myself to a dat manager so that I have rights? At present I am a coordinator.

 

Just checking that it will cause no problems if I do it.

 

Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG I would be happy to do that. However I would like to do the first test together.

It need to be tested by specifying a district to be copied to another county and how its pieces and their civil parishes are to be distributed.

Happy to do 1944B	Keynsham but you need to specify which civil parishes are to go to GLS.

**Comment by geoffj-FUG** [comment]
Kirk

 


1944B

Keynsham

Bitton Holy Trinity, Brislington, Burnet, Compton Dando, Corston, Keynsham, Newton, Newton St Loe, Oldland, Priston, Queen Charlton, Saltford, Stanton Prior, Whitchurch

 

GLS

Bitton Holy Trinity

Oldland

 

SOM

Brislington

Burnett (note correct spelling)

Compton Dando

Corston

Keynsham

Newton

Newton St Loe

Priston

Queen Charlton

Saltford

Stanton Prior

Whitchurch

 

Geoff

**Comment by geoffj-FUG** [comment]
Kirk

 

I have been looking at the proposed Piece to test.

 

It is an 1851 piece and the breaking down of the Piece numbers in the PARMS has meant that there is a neat package of one new style Piece to transfer. I do not have access to GLS so cannot examine the result.

 

May I suggest that we use


RG10/2357  

DULVERTON

Morebath (Devon), Brushford, Dulverton (3), Brompton Regis, Bury

 

This Piece has Morebath in Devon and the other Civil Parishes in Somerset. It therefore needs duplicating and editing. 

 

I also have access to both Counties on test3.

 

Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG  @PatReynolds The following describes the process I used on my computer to do the copy of your civil parish from Somerset to Devon' It is long and tedious; but works. (10 pages step by step in the attached) (an hour)

There are a couple of additions I want to add for locating specific districts pieces and civil parishes rather than the look through a multipage  index. Then I will make you data manager and you can try for yourself.

https://drive.google.com/file/d/18CDltGpUhdLTnGu8hXx2CxJbPTQ3hJms/view?usp=sharing

**Comment by geoffj-FUG** [comment]
Kirk

 

I am going through your procedure and following it on paper.

 

The procedure is at District Level.

 

The District of Dulverton contains 2 Pieces in 1871:


RG10/2357  

DULVERTON

Morebath (Devon), Brushford, Dulverton (3), Brompton Regis, Bury


RG10/2358 

DULVERTON

Skilgate, Huish Champflower, Upton, Exton, Winsford, Exford, Withypoole, Hawkridge, Dulverton

 

Following the theoretical logic of the procedure if we Copy District Dulverton as in the procedure we move both Pieces. We only need to move one.

 

Why are we not working at Sub-District level?

 

Geoff

**Comment by geoffj-FUG** [comment]
Kirk

 

Ahh, the penny has dropped. We need to create the District in the new County as well as the Piece!

Forget the last question.

Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG your penny is worth a great deal more and spot on. 
It might be worth my time to add a bulk delete of a piece if we are going to do this for many Districts

**Comment by geoffj-FUG** [comment]
Kirk

If you do that it should be a soft delete incase a mistake is made at some time.

When the deletes are being done one at a time there is more thought put into it. A hard bulk delete in a hurry can be disasterous!

Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG fully concur will not action

**Comment by geoffj-FUG** [comment]
Kirk
Now that I have got my head around are we ready for me to upgrade myself to Data Manager and test with a different piece?
Just dont want to cause any problems at this juncture.
Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG You are upgraded to Data Manager

**Comment by geoffj-FUG** [comment]
Thanks Kirk

 

Geoff

**Comment by geoffj-FUG** [comment]
Kirk

I have managed to transfer a District and Delete an unwanted transferred one. I have used a different sequence of Actions to get there to your instructions, so there is more than one way through the maze! I will send you a copy.

 

I chose to use Wellington District 1911. There are 20 pieces in it and in real life every one of them will need to be edited!!! 10 stay in Somerset and 10 move to Devon.

 

This also raises another problem. I already have these pieces transcribed in SOM. If I move them to DEV in the PARMS as is required then I lose the ability to manage them as I only have Coordinator access to SOM. So can counties have more than one administrator? It involves a many to many relationship between counties and coordinators so I suspect not.

 

I am confused. Help!!!

I am beginning to lose track of my logic here. If a piece in DEV PARMS is transcribed by a SOM transcriber then it resides in the SOM transcribers CSVProc folder. The DEV Coordinator cannot manage it as the transcriber does not belong to DEV. However the SOM Coordinator cannot manage it as the piece is in DEV in the PARMS. Do we have a catch 22?

 

Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG @PatReynolds So lets split this discussion into 2 parts.

Firstly you have confirmed the software provides the capability to copy a district, its pieces and civil parishes to another county and have confirmed also the ability to remove civil parishes and pieces from a duplicated district #1009. Hence from that perspective we have done what we set out to do in this story.
However it is clear you did not complete the process. Specifically **Now we need to edit the piece to have a different piece number and add a link from the civil parish to a freecen2 place** Without making this change we cannot function correctly.

Thank you for the alternate way of proceeding through the move; I will update the script **if we decide that it is going to be used**. This will be based on the resolution of the second part of the discussion. that follows.

Secondly you are raising concerns about the actual ability to use the capability in an operational manner. To assist in this determination the following are facts. Lets distinguish between people and data
All people (userids) belong to a single syndicate. There are syndicate coordinators who look after the people of the syndicate. A given userid can manage multiple syndicates. But a syndicate can only have one coordinator. Syndicates can work on any data assigned to them regardless of the county. Freebmd for example has many syndicates working on different counties but each working on a different  year. Freereg has a few syndicates working on a single county. Freecen has tended to have a one for one correspondence. ie a syndicate and county are treated as one and the same. However organization is a management practice and not a system one.
Data in the database is organized by counties. A given userid can manage multiple counties. Data managers and system administrators can manage all counties. But a county can only have one coordinator. In this regard county coordinators, data managers are managing slices of the data in the database.
If we were to consider the database as a matrix. Counties are vertical columns and people are the horizontal lines. A contribution of data is provided for the specific intercept.
So if a SOM syndicate transcriber works on a piece for DEV **they will be using the piece number for the Devon version** (Which is why the piece number change is so important in the move). When they upload their records they will be in their folder on the serer BUT will be included in the Devon county coordinators Devon files. Just as my uploads of SOM test pieces appear in your SOM county files even though I am in a different syndicate (called Technical). 
The Devon coordinator would then have either work through you or directly with the transcriber. This is why we have the Display Userids action for coordinators and why I declined to move it within the syndicate. It enables any coordinator to look up the email address for any member. Any member has the ability to use the Communicate action to contact any manager within the system.
I recognize this is a very different way of working and may not be acceptable. 
I hope this answers your questions and removes at least enough of the confusion.

If it felt that is is an unacceptable process then we need to go back to the original requirements that led us to this solution and see if there is an alternate

**Comment by geoffj-FUG** [comment]
Thanks Kirk

 

You are right, I did not carry through to the next stage. That is because my mind was offering challenges to the application of what was happening. I am a learner by doer. Rote learning has never been my forte. As I didn’t quite understand what I was doing, I stopped.

 

I now understand the difference between a Syndicate Coordinator and a County Coordinator. Up to now I have considered them to be one and the same. But they are not. In Business terms the difference is line control and functional control. It now makes sense.

 

Your explanation resolves some of the challenges to my thinking so I will now continue with the process in testing.

 

Geoff

**Comment by geoffj-FUG** [comment]
Kirk

 

I have now successfully completed the process for RG14_14199. It is now RG14_14199A in DEV.

Now I have to round off the instructions.

If the piece is already transcribed the original RG14_14199 will need to be Downloaded.
That part of the transcription referring to Civil Parishes in the new County (in this case DEV) would be Renamed as RG14_14199A. I would suggest that this is initially Uploaded to the original transcriber so that the record chain remains intact. That part of the Original transcription referring to Civil Parishes the original County would be retained in RG14_14199.
The DEV Coordinator will receive a test report on the DEV part and the SOM Coordinator will receive a test report on the SOM part.

Is that correct. I cannot see any other way of achieving this.

Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG 
You are correct

**Comment by geoffj-FUG** [comment]
Kirk

I am writing the instructions for this. Which method should I include - your original version or the way I did it?

Geoff

**Comment by Captainkirkdawson** [comment]
Your way
Kirk Dawson
Programmer and Technical Advisor
FreeUKGEN

**Comment by geoffj-FUG** [comment]
Instructions in progress.
Geoff

**Comment by geoffj-FUG** [comment]
The instructions are now included in the latest version of the Full Handbook on test3.
Geoff

**Comment by geoffj-FUG** [comment]
As we set up the new system we have a need to identify the pieces that have a parish(es) in a different county. As a Coordinator I know which pieces are the problem pieces in my County. But this is not necessarily always the case.
There will be a definite need to identify the relevant pieces. Short of going through every piece in England and Wales is there any method of identifying these pieces? I have tried to look at Districts down to civil parishes but as they are examined one by one and the county info has been stripped this is not possible.
Is there any method of generating  a list of Parishes from the original data that was raked from TNA?
Alternatively can we run run a county / civil parish test against the Place Names Database and get a list of mis-matches?
These mis-matches will be the ones that Rhoda will need to transfer to a different county. If we have a list she will be able to work through them.
In short, we have the process but not the means of identifying the Pieces and Parishes to be transferred
Geoff
